src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanServiceTest.kt

```diff
@@ -207,11 +217,18 @@ internal class ActionPlanServiceTest {
         }
       )
     ).thenReturn(SampleData.sampleActionPlan())
+    whenever(authUserRepository.save(any())).thenReturn(SampleData.sampleAuthUser())
+    whenever(actionPlanAppointmentRepository.save(any())).thenReturn(
+      SampleData.sampleActionPlanAppointment(
+        actionPlan = SampleData.sampleActionPlan(), createdBy = SampleData.sampleAuthUser()
+      )
+    )
 
     val submittedActionPlan = actionPlanService.submitDraftActionPlan(actionPlanId, authUser)
 
     assertThat(submittedActionPlan).isNotNull
     verify(actionPlanValidator).validateSubmittedActionPlan(any())
+    verify(actionPlanAppointmentRepository, times(2)).save(any())
```

Comment:
it's somewhat of a shame we can't validate that the two records are actually created against the current action plan here. could we use argument captors?